src/main/java/com/iexec/resultproxy/jwt/JwtService.java

```diff
@@ -27,37 +54,65 @@ public String getOrCreateJwt(SignedChallenge signedChallenge) {
         return contextualLockRunner.applyWithLock(walletAddress, this::getOrCreateJwt);
     }
 
+    private byte[] initKey(String jwtKeyPath) throws IOException {
+        Path path = Path.of(jwtKeyPath);
+        if (!path.toFile().exists()) {
+            SecureRandom random = new SecureRandom();
+            byte[] bytes = new byte[KEY_SIZE];
+            random.nextBytes(bytes);
+            String content = Base64.getEncoder().encodeToString(bytes);
+            FileHelper.createFileWithContent(jwtKeyPath, content);
+        }
+        String payload = Files.readString(path);
+        return Base64.getDecoder().decode(payload);
+    }
```

Comment:
To be sure about the behavior of these JWT: do we write the key to a file in order to be able to recover from a restart of the Result Proxy?
Wouldn't that be better to write it to the database?

Could you maybe add some comments/JavaDoc so the answer will not be forgotten? :)

Reply:
It is exactly for this purpose and to avoid to invalidate all existing tokens by generating a new key on a restart.